conf/app.routes

```diff
@@ -12,13 +12,14 @@
 #   502:
 #     description: map all upstream error codes to 502
 ###
-GET        /channel-preferences/preference/:channel     uk.gov.hmrc.channelpreferences.controllers.PreferenceController.preference(channel: Channel, enrolmentKey: String, taxIdName: String, taxIdValue: String)
-POST       /channel-preferences/enrolment               @uk.gov.hmrc.channelpreferences.controllers.PreferenceController.enrolment()
-POST       /channel-preferences/process/bounce          @uk.gov.hmrc.channelpreferences.controllers.PreferenceController.processBounce()
+GET        /channel-preferences/preference/:channel         uk.gov.hmrc.channelpreferences.controllers.PreferenceController.preference(channel: Channel, enrolmentKey: String, taxIdName: String, taxIdValue: String)
+POST       /channel-preferences/enrolment                   @uk.gov.hmrc.channelpreferences.controllers.PreferenceController.enrolment()
+POST       /channel-preferences/process/bounce              @uk.gov.hmrc.channelpreferences.controllers.PreferenceController.processBounce()
+POST       /channel-preferences/preference/:key/:status     uk.gov.hmrc.channelpreferences.controllers.PreferenceController.update(key: String, status: String)
```

Comment:
ccan we change the signature of update() to `update(key: Boolean, status: String)` instead of `update(key: String, status: String)`? that will allow you to get rid of `handleItsaStatus(...)` in the controller.

Reply:
So this value won't be always be a `boolean` (string),  for other key / regime it might take another value (any arbitrary string), hence the bit of logic to parse it, and to be able to allow for some extension in the future

Reply:
yes I imagined that might be the case, but because you're using `handleItsaStatus` to guard against values that are neither true or false, which then raises an error, you can simply do that at the HTTP filter level when the route tries to deserialise the request query params which is boolean. As it's unable to deserialize to a boolean, an error is thrown (a badrequest). You can either have a custom error raised or just let play handle it, up to you.

Reply:
unless it's needed today, I wouldn't worry about what the future holds.

Reply:
your custom class could look like this to handle values outside of Boolean: https://github.com/hmrc/channel-preferences/blob/master/app/uk/gov/hmrc/channelpreferences/hub/cds/model/Channel.scala